dlrover/python/master/shard/streaming_dataset_manager.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2022 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import time
+
+from dlrover.python.common.log_utils import default_logger as logger
+from dlrover.python.master.shard.base_dataset_manager import (
+    DatasetManger,
+    DatasetShardCheckpoint,
+    DoingTask,
+    Task,
+)
+from dlrover.python.master.shard.dataset_splitter import (
+    DatasetSplitter,
+    Shard,
+    StreamingDatasetSplitter,
+)
+
+_MAX_TASK_RETRIES = 3
+
+
+class StreamingDatasetManager(DatasetManger):
+    """StreamingDatasetManager create tasks with shards in a dynamic dataset.
+    Attributes:
+        task_type: the type of computation task like "training",
+            "evaluation" and "prediction".
+        batch_size: the size of a batch.
+        dataset_splitter: DatasetSplitter instance to split the dataset
+            into shards.
+    """
+
+    def __init__(
+        self,
+        task_type,
+        batch_size,
+        dataset_splitter: DatasetSplitter,
```

Comment:
`DatasetSplitter` -> `StreamingDatasetSplitter` if the `StreamingDatasetManager` can only use the `StreamingDatasetSplitter`.

Reply:
done